ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -755,10 +755,33 @@ - (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childCompone
 
 - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
 {
+  [self setViewToSnapshot:(RNSScreenStackHeaderSubview *)childComponentView];
   [_reactSubviews removeObject:(RNSScreenStackHeaderSubview *)childComponentView];
   [childComponentView removeFromSuperview];
 }
 
+- (void)setViewToSnapshot:(RNSScreenStackHeaderSubview *)childComponentView
+{
+  UINavigationItem *navitem = _screenView.controller.navigationItem;
+  UIView *snapshot = [childComponentView snapshotViewAfterScreenUpdates:NO];
+
+  switch (childComponentView.type) {
+    case RNSScreenStackHeaderSubviewTypeLeft:
+      navitem.leftBarButtonItem.customView = snapshot;
+      break;
+    case RNSScreenStackHeaderSubviewTypeCenter:
+    case RNSScreenStackHeaderSubviewTypeTitle:
+      navitem.titleView = snapshot;
+      break;
+    case RNSScreenStackHeaderSubviewTypeRight:
+      navitem.rightBarButtonItem.customView = snapshot;
+      break;
+    case RNSScreenStackHeaderSubviewTypeSearchBar:
+    case RNSScreenStackHeaderSubviewTypeBackButton:
+      break;
+  }
+}
```

Comment:
I wanted to suggest to refactor out this code, so that it is shared with the `+ [RNSScreenStackHeaderConfig updateViewController:]` method, but it seems that we need only a subset of the functionality there. 

I would like you to add comments here, and in appropriate fragment of the `+ [RNSScreenStackHeaderConfig updateViewController:]` method that these two must be kept in sync.

Reply:
https://github.com/software-mansion/react-native-screens/pull/2247/commits/1f55775e0778460448b70773e6675a6e7a19bb88